### PR TITLE
1. Add test files

### PR DESCRIPTION
**Pull request by Zel-hub7**
perfo-1
perfo-2

1. I used  Capybara to write integration tests for each view in the project.
2. I have written and passed all the integration test.
3. I fixed n + 1 problems

### CODE REVIEW

File: spec/integration/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show', type: :feature do
```
Comment (V-Blaze) [inline spec/integration/user_show_spec.rb:3]: - You have done great so far :+1: :100: But there are a few requirements you need to meet before moving to the next milestone. :smile: 

- - Screenshot reference:
Selection_032

- [ ] Your `user_show` spec is missing one of the required test cases. According to the requirement, there should be `Tests that redirect to the user's post's index page when you click on a see all posts.`. This is currently not implemented and I suggest you add the test case :+1: :100: :smile:

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# spec/features/post_index_spec.rb
+
+require 'rails_helper'
+
```
Comment (V-Blaze) [inline spec/integration/post_index_spec.rb:4]: -  Screenshot reference:
Selection_033

- [ ] In the users `Post_index` there needs to be a `Test that asserts you can see a section for pagination if there are more posts than fit on the view.`. The pagination tests were not implemented and I suggest you add it as it is a part of the requirement. :smile: :wave: :small_airplane:

File: spec/integration/post_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# spec/integration/post_show_spec.rb
+
+require 'rails_helper'
```
Comment (V-Blaze) [inline spec/integration/post_show_spec.rb:3]: - Screenshot reference:
Selection_034

- [ ] According to the project requirement, you are currently missing two test cases, a `Test that asserts you can see who wrote the post.` and a `Test that asserts you can see the username of each commentor.` I suggest you include these test cases according to the project requirements. :smile: :waffle: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Zel-hub7, @AlyAtalla  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- All written tests are passing✔
- Fixed `N+1 problem` correctly 👋 
-  Well documented codebase ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by LionRouge1 — APPROVED** [review]
Hi @Zel-hub7 & @AlyAtalla, 

# STATUS: APPROVED! 🥇 💯 
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

done

## Highlights
- Add all the missing test cases as suggested by the reviewer :heavy_check_mark: 
- Nice work you meet all the requirements :1st_place_medal: 
- All the tests are passing :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**